packages/hash/frontend/src/components/layout/PageHeader/SearchBar/SearchInput.tsx

```diff
@@ -0,0 +1,179 @@
+import {
+  Box,
+  IconButton,
+  InputAdornment,
+  OutlinedInput,
+  Tooltip,
+  Typography,
+} from "@mui/material";
+import { useCallback, useEffect, useRef } from "react";
+import { useKeys } from "rooks";
+import { faXmark } from "@fortawesome/free-solid-svg-icons";
+
+import { SearchIcon, FontAwesomeSvgIcon } from "../../../icons";
+
+const ClearSearchIcon: React.FC<{
+  clearSearch: () => void;
+}> = ({ clearSearch }) => {
+  return (
+    <Box
+      sx={{
+        display: "flex",
+        height: "100%",
+        alignItems: "center",
+        cursor: "pointer",
+        fontWeight: "bold",
+        width: "26px",
+        justifyContent: "flex-end",
+        marginRight: 1,
+      }}
+    >
+      <Tooltip title="Clear search" placement="right">
+        <IconButton
+          sx={(theme) => ({
+            width: theme.spacing(2.5),
+            height: theme.spacing(2.5),
+            display: "flex",
+            alignItems: "center",
+            justifyContent: "center",
+            // @todo figure out how to fetch this from `theme`
+            borderRadius: "2px",
+            color: theme.palette.gray[50],
+
+            "&:hover": {
+              transition: theme.transitions.create([
+                "color",
+                "background-color",
+              ]),
+              backgroundColor: theme.palette.gray[20],
+              color: theme.palette.gray[80],
+            },
+          })}
+          onClick={clearSearch}
+        >
+          <FontAwesomeSvgIcon icon={faXmark} />
+        </IconButton>
+      </Tooltip>
+    </Box>
+  );
+};
+
+const ShortcutIcon = () => {
+  const isMac = navigator.userAgent.toUpperCase().includes("MAC");
+
+  return (
+    <Box
+      sx={{
+        height: "100%",
+        pointerEvents: "none",
+        display: "flex",
+        alignItems: "center",
+        fontWeight: "bold",
+        marginRight: 0.5,
+      }}
+    >
+      <Box
+        sx={(theme) => ({
+          height: "30px",
+          display: "flex",
+          alignItems: "center",
+          justifyContent: "center",
+          color: theme.palette.gray[50],
+          backgroundColor: theme.palette.gray[20],
+          borderRadius: "2px",
+          px: 1,
+          py: 0.75,
+        })}
+      >
+        <Typography variant="microText">
+          {isMac ? "Cmd" : "Ctrl"} + P
+        </Typography>
+      </Box>
+    </Box>
+  );
+};
+
+export const SearchInput: React.FC<{
+  displayedQuery: string;
+  isMobile: boolean;
+  setResultListVisible: (visible: boolean) => void;
+  setQueryText: (queryText: string) => void;
+}> = ({ displayedQuery, isMobile, setResultListVisible, setQueryText }) => {
+  const isMac = navigator.userAgent.toUpperCase().includes("MAC");
```

Comment:
Doesn't have to be done as part of this PR, but I think that's a good opportunity for a react hook so that we can reuse this logic and state in various places throughout the app.